container/rkt/handler.go

```diff
@@ -0,0 +1,488 @@
+	// Absolute path to the cgroup hierarchies of this container.
+	// (e.g.: "cpu" -> "/sys/fs/cgroup/cpu/test")
+	cgroupPaths map[string]string
+
+	// Manager of this container's cgroups.
+	cgroupManager cgroups.Manager
+
+	// Whether this container has network isolation enabled.
+	hasNetwork bool
+
+	fsInfo         fs.FsInfo
+	externalMounts []common.Mount
+
+	rootFs string
+
+	isPod bool
+
+	aliases []string
+
+	pid int
+
+	rootfsStorageDir string
+
+	// Filesystem handler.
+	fsHandler common.FsHandler
+
+	ignoreMetrics container.MetricSet
+}
+
+func (self *rktContainerHandler) GetCgroupPaths() map[string]string {
+	return self.cgroupPaths
+}
+
+func (self *rktContainerHandler) GetMachineInfoFactory() info.MachineInfoFactory {
+	return self.machineInfoFactory
+}
+
+func (self *rktContainerHandler) GetName() string {
+	return self.name
+}
+
+func (self *rktContainerHandler) GetExternalMounts() []common.Mount {
+	return self.externalMounts
+}
+
+func (self *rktContainerHandler) HasNetwork() bool {
+	return self.hasNetwork && !self.ignoreMetrics.Has(container.NetworkUsageMetrics)
+}
+
+func (self *rktContainerHandler) HasFilesystem() bool {
+	if !self.ignoreMetrics.Has(container.DiskUsageMetrics) {
+		return true
+	}
+	return false
+}
+
+func newRktContainerHandler(name string, cgroupSubsystems *libcontainer.CgroupSubsystems, machineInfoFactory info.MachineInfoFactory, fsInfo fs.FsInfo, watcher *common.InotifyWatcher, rootFs string, ignoreMetrics container.MetricSet) (container.ContainerHandler, error) {
+	rktPath := "/var/lib/rkt"
+
+	glog.Infof("rkt cgroup name = %q", name)
+	aliases := make([]string, 1)
+	isPod := false
+
+	parsed, err := parseName(name)
+	if err != nil {
+		return nil, fmt.Errorf("this should be impossible!, new handler failing, but factory allowed, name = %s", name)
+	}
+
+	if parsed.Container == "" {
+		isPod = true
+		aliases = append(aliases, parsed.Pod)
+	} else {
+		aliases = append(aliases, parsed.Pod+":"+parsed.Container)
+	}
+
+	cgroupPaths := common.MakeCgroupPaths(cgroupSubsystems.MountPoints, name)
+
+	cHints, err := common.GetContainerHintsFromFile(*common.ArgContainerHints)
+	if err != nil {
+		return nil, err
+	}
+
+	// Generate the equivalent cgroup manager for this container.
+	cgroupManager := &cgroupfs.Manager{
+		Cgroups: &configs.Cgroup{
+			Name: name,
+		},
+		Paths: cgroupPaths,
+	}
+
+	pid := os.Getpid()
+	hasNetwork := false
+	if isPod {
+		hasNetwork = true
+
+		pids, err := cgroups.GetPids(cgroupPaths["cpu"])
+		if err == nil {
+			pid = pids[0]
+		} else {
+			glog.Infof("couldn't get processes as %v", err)
+		}
+	}
+
+	//SJP: unsure the point of this code, if it event does anything today?
+	var externalMounts []common.Mount
+	for _, container := range cHints.AllHosts {
+		if name == container.FullName {
+			/*libcontainerState.NetworkState = network.NetworkState{
+				VethHost:  container.NetworkInterface.VethHost,
+				VethChild: container.NetworkInterface.VethChild,
+			}
+			hasNetwork = true*/
```

Comment:
This PR needs to include the metrics selection (opt-out) patch that I posted recently.